core/common/src/main/java/alluxio/underfs/ObjectUnderFileSystem.java

```diff
@@ -328,35 +327,15 @@ private void submitBatch() throws IOException {
         int batchNumber = mBatches.size();
         mBatches.add(new ArrayList<>(mCurrentBatchBuffer));
         mCurrentBatchBuffer.clear();
-        mBatchesResult.add(batchNumber,
-            mExecutorService.submit(new OperationThread(mBatches.get(batchNumber))));
-      }
-    }
-
-    /**
-     * Thread class to operate on a batch of objects.
-     */
-    @NotThreadSafe
-    protected class OperationThread implements Callable<List<T>> {
-      List<T> mBatch;
-
-      /**
-       * Operate on a batch of inputs.
-       *
-       * @param batch a list of inputs for the current batch
-       */
-      public OperationThread(List<T> batch) {
-        mBatch = batch;
-      }
-
-      @Override
-      public List<T> call() {
-        try {
-          return operate(mBatch);
-        } catch (IOException e) {
-          // Do not append to success list
-          return Collections.emptyList();
-        }
+        List<T> batch = mBatches.get(batchNumber);
+        mBatchesResult.add(batchNumber, mExecutorService.submit(() -> {
+          try {
+            return operate(batch);
+          } catch (IOException e) {
+            // Do not append to success list
+            return Collections.emptyList();
+          }
+        }));
```

Comment:
unrelated to your PR, but why is mBatches needed here?  and to make the code compact and to the point i made before about synchronized identifier, one list (mCurrentBatchBuffer) is enough to submit to threadpool and lock could be taken around that during modification/access is it?